### ISSUE TITLE
Indexed getter for readonly CSSStyleDeclaration is broken

### ISSUE THREAD

**Issue by jdm** [servo/servo#26178]
`window.getComputedStyle(document.documentElement)[0]` returns `align-content` in Firefox, but `undefined` in Servo. `get_property_value` [correctly](https://github.com/servo/servo/blob/master/components/script/dom/cssstyledeclaration.rs#L256-L259) has a different codepath for readonly (ie. computed) style declarations, but [IndexedGetter](https://github.com/servo/servo/blob/7ae11588dc36e644750008d1224a6d93c9e49d08/components/script/dom/cssstyledeclaration.rs#L425-L426) (and Length) do not. It shouldn't be _too_ hard to reuse the get_computed_style code path for the indexed getter as long as we can figure out how to map the array indices to property names, but it's not totally clear how to get the length of the computed styles.

This also has the effect that `Object.getOwnPropertyNames(window.getComputedStyle(document.documentElement))` returns an empty object, and this ends up breaking the MotionMark benchmark's browser detection.

**Comment by jdm** [comment]
This should be less work than I initially thought, since https://github.com/servo/servo/pull/26120/commits/c10e8399246e5254a080c2dc809d1c180f06cc93 made it easier to access style data for a node. We'll need the following:
* an equivalent for https://searchfox.org/mozilla-central/source/layout/style/GenerateComputedDOMStyleGenerated.py that generates a map of indexes to CSS properties
* use that generated map for the IndexedGetter and Length methods in CSSStyleDeclaration for readonly declarations

It probably makes sense to generate this map in the same way that we generate CSSStyleDeclaration.webidl, based on the output of the style crate's CSS property generation.

**Comment by lumiscosity** [comment]
This is fixed now, running a `console.log()` on the first snippet returns `align-content` and on the second snippet returns `["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "10", "11", "12", "13", "14", …]`.


Tested on Arch Linux x86_64, Servo built against commit 9d29017c0dc6eec4ca7cd868cc13056e0bc0c64a, profiling build.

**Comment by jdm** [comment]
Yep, this was fixed by https://github.com/servo/servo/pull/31499.
